### ISSUE TITLE
Enrichment analysis fisher test issue

### ISSUE THREAD

**Issue by Daniel-Calvez** [harryhaller001/keggtools#12]
### Report

The enrichment analysis (Enrichment.run_analysis()) fails. The error is caused by a negative value (d_var) in the matrix used for the Fisher test.

### Version information

Python 3.14.2
Package                   Version
------------------------- -----------
adjustText                1.3.0
anndata                   0.12.9
annotated-types           0.7.0
anyio                     4.12.1
archspec                  0.2.5
argcomplete               3.6.3
argon2-cffi               25.1.0
argon2-cffi-bindings      25.1.0
array-api-compat          1.13.0
arrow                     1.4.0
asttokens                 3.0.1
async-lru                 2.1.0
attrs                     25.4.0
babel                     2.17.0
bcrypt                    5.0.0
beautifulsoup4            4.14.3
bleach                    6.3.0
boltons                   25.0.0
bonesistools              1.1.3
Brotli                    1.2.0
cached-property           1.5.2
cattrs                    26.1.0
certifi                   2026.1.4
cffi                      2.0.0
chardet                   5.2.0
charset-normalizer        3.4.4
click                     8.3.1
colorama                  0.4.6
comm                      0.2.3
conda                     26.1.0
conda-build               26.1.0
conda_index               0.7.0
conda-libmamba-solver     25.11.0
conda-package-handling    2.4.0
conda_package_streaming   0.12.0
contourpy                 1.3.3
cryptography              45.0.7
cycler                    0.12.1
dcor                      0.6
debugpy                   1.8.20
decorator                 5.2.1
decoupler                 2.1.4
defusedxml                0.7.1
Deprecated                1.3.1
dill                      0.4.1
distro                    1.9.0
docopt-ng                 0.9.0
docrep                    0.3.2
donfig                    0.8.1.post1
et_xmlfile                2.0.0
evalidate                 2.0.5
exceptiongroup            1.3.1
executing                 2.2.1
face                      24.0.0
fast-array-utils          1.3.1
fastjsonschema            2.21.2
filelock                  3.24.2
fonttools                 4.61.1
formulaic                 1.2.1
formulaic-contrasts       1.0.0
fqdn                      1.5.1
frozendict                2.4.7
ftpretty                  0.4.0
future                    1.0.0
glom                      25.12.0
goatools                  1.5.2
google-crc32c             1.8.0
h11                       0.16.0
h2                        4.3.0
h5py                      3.15.1
hpack                     4.1.0
httpcore                  1.0.9
httpx                     0.28.1
hyperframe                6.1.0
idna                      3.11
igraph                    1.0.0
importlib_metadata        8.7.0
importlib_resources       6.5.2
inflect                   7.5.0
interface_meta            1.3.0
invoke                    2.2.1
ipykernel                 7.1.0
ipython                   9.9.0
ipython_pygments_lexers   1.1.1
ipywidgets                8.1.8
isoduration               20.11.0
jedi                      0.19.2
Jinja2                    3.1.6
joblib                    1.5.3
json5                     0.13.0
jsonpatch                 1.33
jsonpointer               3.0.0
jsonschema                4.26.0
jsonschema-specifications 2025.9.1
jupyter                   1.1.1
jupyter_client            8.8.0
jupyter-console           6.6.3
jupyter_core              5.9.1
jupyter-events            0.12.0
jupyter-lsp               2.3.0
jupyter_server            2.17.0
jupyter_server_terminals  0.5.4
jupyterlab                4.5.3
jupyterlab_pygments       0.3.0
jupyterlab_server         2.28.0
jupyterlab_widgets        3.0.16
keggtools                 1.1.0
kiwisolver                1.4.9
lark                      1.3.1
legacy-api-wrap           1.5
legendkit                 0.3.4
libarchive-c              5.3
libmambapy                2.5.0
lief                      0.17.0
llvmlite                  0.46.0
lxml                      6.0.2
markdown-it-py            4.0.0
MarkupSafe                3.0.3
marsilea                  0.5.6
matplotlib                3.10.8
matplotlib-inline         0.2.1
mdurl                     0.1.2
menuinst                  2.4.2
mistune                   3.2.0
more-itertools            10.8.0
msgpack                   1.1.2
munkres                   1.1.4
narwhals                  2.15.0
natsort                   8.4.0
nbclient                  0.10.4
nbconvert                 7.17.0
nbformat                  5.10.4
nest_asyncio              1.6.0
networkx                  3.6.1
notebook                  7.5.3
notebook_shim             0.2.4
numba                     0.63.1
numcodecs                 0.16.5
numpy                     2.3.5
omnipath                  1.0.8
openpyxl                  3.1.5
overrides                 7.7.0
packaging                 26.0
pandas                    2.3.3
pandocfilters             1.5.0
paramiko                  4.0.0
parso                     0.8.5
patsy                     1.0.2
pexpect                   4.9.0
pillow                    12.1.0
pip                       25.3
pkginfo                   1.12.1.2
platformdirs              4.5.1
pluggy                    1.6.0
prometheus_client         0.24.1
prompt_toolkit            3.0.52
psutil                    7.2.2
ptyprocess                0.7.0
pure_eval                 0.2.3
pybiomart                 0.2.0
pycosat                   0.6.6
pycparser                 3.0
pycurl                    7.45.7
pydantic                  2.12.5
pydantic_core             2.41.5
pydantic-xml              2.19.0
pydeseq2                  0.5.4
pydot                     4.0.1
Pygments                  2.19.2
PyNaCl                    1.5.0
pynndescent               0.5.13
pyparsing                 3.3.2
pypath_common             0.2.6
pypath-omnipath           0.16.20
pyreadr                   0.5.4
pysftp                    0.2.9
PySide6                   6.10.1
PySocks                   1.7.1
python-dateutil           2.9.0.post0
python-json-logger        2.0.7
pytz                      2025.2
PyYAML                    6.0.3
pyzmq                     27.1.0
rdata                     1.0.0
referencing               0.37.0
requests                  2.32.5
requests-cache            1.3.0
rfc3339_validator         0.1.4
rfc3986-validator         0.1.1
rfc3987-syntax            1.1.0
rich                      14.3.2
rpds-py                   0.30.0
rpy2                      3.6.4
rpy2-rinterface           3.6.3
rpy2-robjects             3.6.3
ruamel.yaml               0.18.17
ruamel.yaml.clib          0.2.15
scanpy                    1.12
scikit-learn              1.8.0
scipy                     1.16.3
seaborn                   0.13.2
Send2Trash                2.1.0
session-info              1.0.0
session-info2             0.3
setuptools                80.10.2
shiboken6                 6.10.1
six                       1.17.0
sniffio                   1.3.1
soupsieve                 2.8.3
sqlparse                  0.5.5
stack_data                0.6.3
statsmodels               0.14.6
stdlib-list               0.12.0
tabulate                  0.9.0
terminado                 0.18.1
texttable                 1.7.0
threadpoolctl             3.6.0
timeloop                  1.0.2
tinycss2                  1.5.1
toml                      0.10.2
tomli                     2.4.0
tomlkit                   0.14.0
tornado                   6.5.4
tqdm                      4.67.1
traitlets                 5.14.3
truststore                0.10.4
typeguard                 4.4.4
typing_extensions         4.15.0
typing-inspection         0.4.2
typing_utils              0.1.0
tzdata                    2025.3
tzlocal                   5.3.1
umap-learn                0.5.11
unicodedata2              17.0.0
uri-template              1.3.0
url-normalize             2.2.1
urllib3                   2.6.3
wcwidth                   0.5.2
webcolors                 25.10.0
webencodings              0.5.1
websocket-client          1.9.0
wget                      3.2
widgetsnbextension        4.0.15
wrapt                     2.0.1
xarray                    2026.1.0
xgboost                   3.1.3
xlrd                      2.0.2
xlsxwriter                3.2.9
xmltodict                 1.0.2
yq                        3.4.3
zarr                      3.1.5
zipp                      3.23.0
zstandard                 0.25.0

**Comment by harryhaller001** [comment]
Hi, thanks for your bug report. Could you provide the error message and if possible code to reproduce the issue?
